vspace/src/main/java/edu/asu/diging/vspace/web/staff/SpaceOverviewController.java

```diff
@@ -0,0 +1,317 @@
+     * @param space            passing the space object
+     * @param node             creating json for space node in the spaceoverview
+     *                         graph
+     * @param contextPath      This variable holds the contextpath of the
+     *                         application
+     * @param spaceId          This variable is holding the space id corresponding
+     *                         each space in vspace
+     */
+    private void createSpaceNode(StringBuilder linkPathBuilder, StringBuilder imagePathBuilder, boolean isModule,
+            Space space, ObjectNode node, String contextPath, String spaceId) {
+        linkPathBuilder.setLength(0);
+        linkPathBuilder.append(contextPath);
+        linkPathBuilder.append(SpaceController.STAFF_SPACE_PATH);
+        linkPathBuilder.append(spaceId);
+        node.put("link", linkPathBuilder.toString());
+        imagePathBuilder.setLength(0);
+        imagePathBuilder.append(contextPath);
+        imagePathBuilder.append(ImageApiController.API_IMAGE_PATH);
+        imagePathBuilder.append(space.getImage().getId());
+        node.put("img", imagePathBuilder.toString());
+        node.put("isModule", isModule);
+
+        if (space.getSpaceStatus() != null && (space.getSpaceStatus().equals(SpaceStatus.UNPUBLISHED))) {
+            node.put("isUnpublished", true);
+        } else {
+            node.put("isUnpublished", false);
+        }
+        if (space.isHideIncomingLinks()) {
+            node.put("isHideIncomingLinks", true);
+        } else {
+            node.put("isHideIncomingLinks", false);
+        }
+    }
+
+    /**
+     * creating json for module node in the spaceoverview graph
+     * 
+     * @param linkPathBuilder  StringBuilder object to create the URL link for
+     *                         module in staff side
+     * @param imagePathBuilder StringBuilder object to create the URL link for image
+     *                         in staff side
+     * @param isModule         boolean variable to indicate whether it is module or
+     *                         not
+     * @param node             creating json for module node in the spaceoverview
+     *                         graph
+     * @param contextPath      This variable holds the contextpath of the
+     *                         application
+     * @param moduleId         This variable is holding the module id corresponding
+     *                         each module in vspace
+     */
+    private void createModuleNode(StringBuilder linkPathBuilder, StringBuilder imagePathBuilder, boolean isModule,
+            ObjectNode node, String contextPath, String moduleId) {
+        linkPathBuilder.setLength(0);
+        linkPathBuilder.append(contextPath);
+        linkPathBuilder.append(ModuleController.STAFF_MODULE_PATH);
+        linkPathBuilder.append(moduleId);
+        node.put("link", linkPathBuilder.toString());
+        node.put("img", "");
+        node.put("isModule", isModule);
```

Comment:
if this is for creating module nodes, why is there a flag `isModule` passed?

Reply:
That boolean value is necessary in the Json for both space and module. In case of space this flag false and in case of module this flag is true.
Based on the flag in the graph.html some actions are performed.